indra/llcommon/llerror.h

```diff
@@ -90,6 +90,42 @@ const int LL_ERR_NOERR = 0;
 #define llverify(func)			do {if (func) {}} while(0)
 #endif
 
+
+// The following 2 macros are used below in llassert_returning
+#define llvalue_flag0(value)
+#define llvalue_flag1(value) value
+
+// The macro llassert_returning is used in the following macros:
+// - llassert_message_return_value
+// - llassert_message_return
+// - llassert_return_value
+// - llassert_return
+#define llassert_returning(condition, separator, message, flag, value) \
+{                                                                      \
+    if (LL_UNLIKELY(!(condition)))                                     \
+    {                                                                  \
+        LL_WARNS()                                                     \
```

Comment:
basic llassert() is develop build only (llassert_always work everywhere) I think this should be kept the same way due to viewer handling massive amount of images. If something happens viewer will freeze due to excessive logging.

Reply:
I agree. Also, a big use of asserts is to make it easier for developers to spot errors in code, because it triggers a crash breakpoint in the debugger. I think preserving that behavior in a debug build is very useful. This macro is technically not an assert because it doesn't do that.